components/debounce/browser/debounce_rule.cc

```diff
@@ -86,15 +143,44 @@ bool DebounceRule::Apply(const GURL& original_url, GURL* final_url) const {
   if (!include_pattern_set_.MatchesURL(original_url))
     return false;
 
-  std::string unescaped_value;
-  if (!net::GetValueForKeyInQuery(original_url, param_, &unescaped_value))
-    return false;
-  if ((action_ == kDebounceBase64DecodeAndRedirectToParam) &&
-      (!base::Base64UrlDecode(unescaped_value,
-                              base::Base64UrlDecodePolicy::IGNORE_PADDING,
-                              &unescaped_value))) {
+  if (!DebounceRule::CheckPrefForRule(prefs)) {
     return false;
   }
+
+  std::string unescaped_value;
+
+  if (action_ == kDebounceRegex) {
+    // apply param regex to original url.
+    const re2::RE2 kPatternRegex(param_);
```

Comment:
Code style: should be `pattern_regex`, kSomething is allowed only for global constants.